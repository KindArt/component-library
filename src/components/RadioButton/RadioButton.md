Colours:

```tsx
<RadioButton defaultChecked />
<RadioButton defaultChecked colour="BLUE" />
<RadioButton defaultChecked colour="ORANGE" />
<RadioButton defaultChecked colour="RED" />
```

With Text:

```tsx
<RadioButton label="This is a label" description="This is a example description" />
```
