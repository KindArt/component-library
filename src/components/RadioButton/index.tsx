import React, { FC } from 'react';
import Checkbox, { ICheckboxProps } from '../Checkbox';

const RadioButton: FC<ICheckboxProps> = ({ ...rest }) => <Checkbox radio={true} type="radio" {...rest} />;

export default RadioButton;
