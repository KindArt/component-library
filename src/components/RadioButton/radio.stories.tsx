import React from 'react';
import RadioButton from '.';

export default {
  component: RadioButton,
  title: 'Radio Button'
};

export const normal = () => <RadioButton name="normal" />;

export const multiple = () => {
  return (
    <div>
      <RadioButton name="multi-1" />
      <RadioButton name="multi-1" />
      <RadioButton name="multi-1" />
    </div>
  );
};
