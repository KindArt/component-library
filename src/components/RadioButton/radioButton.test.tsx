import React from 'react';
import { shallow } from 'enzyme';
import RadioButton from './';

describe('<RadioButton />', () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = shallow(<RadioButton name="radioButton-test" />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Visuals', () => {
    it('should have the radio prop set to true', () => {
      const actual = wrapper.props().radio;
      const expected = true;
      expect(actual).toEqual(expected);
    });

    it('should have the type prop set to radio', () => {
      const actual = wrapper.props().type;
      const expected = 'radio';
      expect(actual).toEqual(expected);
    });
  });
});
