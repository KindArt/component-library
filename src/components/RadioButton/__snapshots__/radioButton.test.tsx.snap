// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> should render 1`] = `
<Checkbox
  name="radioButton-test"
  radio={true}
  type="radio"
/>
`;
